service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -1,17 +1,120 @@
 package uk.gov.hmcts.reform.fpl.model;
 
-import lombok.AllArgsConstructor;
+import lombok.AccessLevel;
 import lombok.Builder;
 import lombok.Data;
+import lombok.Getter;
+import uk.gov.hmcts.reform.fpl.enums.OrderType;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentSocialWorkOther;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasDocumentStatus;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasEPOGrounds;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasMainApplicant;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasThresholdDetails;
+import uk.gov.hmcts.reform.fpl.validators.interfaces.HasThresholdReason;
 
 import java.util.List;
 
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
 @Data
-@Builder(toBuilder = true)
-@AllArgsConstructor
+@Builder
+@HasEPOGrounds
+@HasThresholdReason
+@HasThresholdDetails
+@HasDocumentStatus
+@HasMainApplicant
+@SuppressWarnings("membername")
 public class CaseData {
+    @NotBlank(message = "Enter a case name")
+    private final String caseName;
+    private final String gatekeeperEmail;
+    private final String caseLocalAuthority;
+    private final Risks risks;
+    @NotNull(message = "You need to add details to orders and directions needed")
+    @Valid
+    private final Orders orders;
+
+    public boolean hasEPOGrounds() {
+        return orders != null && orders.getOrderType() != null
+            && orders.getOrderType().contains(OrderType.EMERGENCY_PROTECTION_ORDER);
+    }
+
+    private final Grounds grounds;
```

Comment:
Maybe its just me but I dont think this reads well it sounds very informal or something. I feel something like "Details needed to be entered for children" or something sounds more official but im not a language expert :)